### DISCUSSION TITLE
@push('scripts') @endpush not working in steps components

### DISCUSSION THREAD

**Discussion by kennethsantianez** [spatie/laravel-livewire-wizard#102] [Q&A]
Hi,

Thank you for an amazing package.

I have set up a wizard component, all is working well. Below code is added in 2nd step component, but it is not loaded.
But if I push scripts from the 1st step component, it is working. What I want is to load custom scripts after 1st step component.

`

	@push('scripts')
		<script>
			$(function() {

				console.log('2nd step');

			})
		</script>
	@endpush

`

In parent layout file, `@stack('scripts')` is already added. Livewire also is completely installed.

**Comment by ju5t** [comment]
That's because you can only `@push` the first time it's rendered. After that, you're using dom-diffing to go to the next step. When that step renders it has no idea where the `@stack` is as it's already been rendered. 

This cannot be solved by the wizard.

**Reply by ju5t** [reply]
@kennethsantianez check Livewire v3.2.0. This may solve your use case with `@assets`. I have not been able to test this version yet myself though.

**Reply by claudio1994-oliveira** [reply]
@assets  solved it for me. 

Fallow the example with select2

`
   @assets
    <script src="https://code.jquery.com/jquery-3.7.1.min.js"
            integrity="sha256-/JqT3SQfawRcv/BIHPThkBvs0OEvtFFmqPF/lYI/Cxo=" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/npm/select2@4.1.0-rc.0/dist/js/select2.min.js"></script>
    @endassets
    @script
    <script>
        $(document).ready(function () {
            $("#countries").select2();

            $("#countries").on("change", function (e) {
                @this.
                set("country", e.target.value);
            });
        });
    </script>
    @endscript

`
